### DISCUSSION TITLE
How is UIPad.Inside supposed to work?

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#554] [Q&A]
I was testing with 

```
UI.WindowBegin(...);
UI.PanelBegin(UIPad.Inside);
<lots of UI elements>
UI.PanelEnd();
UI.WindowEnd();
```

in a `UI.Window` and ended up with a window that kept slowly growing horizontally over time 😄. So I suspect the way I'm using the panel padding isn't entirely correct? I was looking for a way to add slightly more padding to the inside of the Panel, to put a bit more space between the panel edge and the elements within it. 

Edit 2: Okay, so there's `UI.PanelAt()` which allows a specific size, but also needs a 3D position. I suspect this won't really work to put a fixed-sized panel on a Window? Or is there a way to get the current layout 3D position?

Edit: I'm trying to create a window with a number of "tabs", by using a set of `UI.Radio` buttons which control the current tab to show. The `UI.Panel` then contains the current tab's UI elements. This sort of works, but since the panel cannot be given a specific size it's a bit visually unappealing, as switching tabs results in a differently sized panel, hence my tests with the padding setting

**Comment by maluoi** [comment]
The window expanding like that is probably a bug! I might enjoy a repro case if you can isolate it, but I wouldn't sweat too hard on that, next build will have a good number of layout fixes that could help.

There's three modes for the panel, one of them does add padding inside of the panel, while the other two just work with the existing spacing. I'm AFK right now and don't quite remember at the moment which is which :D You can use UI.PanelAt(UI.LayoutAt, UI.LayoutRemaining) to fill the remaining space with a panel, which is handy in some cases!

Next build will also include UI.LayoutPush/Pop, which would allow you to push an isolated chunk of layout onto a layout stack, which I think should help a lot in this case when combined with LayoutReserve! I've also got a UI.LayoutPushCut coming that should change a lot how UI design will work in SK!
